### ISSUE TITLE
Holding Patterns for Space Liner to save Megacity AI Pathing.

### ISSUE THREAD

**Issue by Artytoo** [OpenApoc/OpenApoc#1252]
Game Transtellar Space Liner AI is making pathing issues in Megacity and slowing computers down with AI Pathing.

Solution:
Using a holding pattern around the outside of the Megacity would stop AI Pathing processes.
Point a and point b holding pattern exit points depending on Cityscape.

Alternatives:
I literally took control of the space liners with midnight editor to be flying at the highest level with the most aggression and it still caused AI pathing conflicts with alien invasion active.

Mod?:
I am pretty sure if it was used as a mod, it would get overused.

**Include any design documents**
You can watch the space liners for 25 years and study up on aviation if you like.
Holding patterns are designed to stop these conflicts.

Point a and point b refer to the closest points of the holding pattern to the Transtellar buildings.

If moved in a counter clockwise rotation around Megacity it would mean in superhuman level, 3 sides would go to the largest Transtellar building and one side would go to the smallest Transtellar building.

**Additional context**
Holding pattern are a marvelous thing and their implementation would give rise to some really faster Megacities.

If you pick up sprite tinting you could have the space liner tinted red like klaxons when it was low on fuel as a mod for emergency landing.

**Comment by FilmBoy84** [comment]
Agreed, this is a good solution and the Original Game does something similar when all hangers / landing ports are in use - you will see vehicles circling the property in a defined holding space

There are two issues with stuck vehicles and the impact on pathfinding I'm aware of

1) OpenApoc does not currently recognise or use "free" empty hangers or garages all of the time - there needs to be a check every XX ticks that tries to reroute to a free alternative hanger or garage at the property if the most direct route is blocked

2) Specify a holding area around each property for vehicles to circle until a free space is available - an optimisation, although a significant deviation from the OG, would be to have vehicles enter this holding area first, then travel beyond it only when a hanger / landing tube or garage becomes available

It's definitely something I would love to see considered in resolving pathfinding issues in OA

Thanks for raising this issue 🍻
